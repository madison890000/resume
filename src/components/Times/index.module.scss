.times {
  & > span {
    display: inline-block;
    margin: 0 var(--inline-block-margin);
    min-width: 90px;
    text-align: center;
    font-style: italic;
    font-size: var(--base-font-size-middle);
  }

  & > span:nth-child(2) {
    min-width: 10px;
  }

  @media screen and (max-width: 414px ) {

    & > span {
      min-width: 60px;
    }
  }
}

