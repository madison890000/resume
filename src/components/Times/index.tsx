import { formatDate } from '../../utils/date';
import React from 'react';
import styles from './index.module.scss';

interface TimesProps {
    start: Date;
    end?: Date;
}

const Times = ({ start, end }: TimesProps) => (
    <div className={styles.times}>
        <span>{formatDate(start)}</span>
        <span>-</span>
        <span>{end ? formatDate(end) : 'present'}</span>
    </div>
)

export default Times
