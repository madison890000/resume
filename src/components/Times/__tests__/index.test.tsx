import React from 'react';
import Times from '../index';
import renderer from 'react-test-renderer';

test('renders Times', () => {
    const start = new Date('2012-01-01');
    const end = new Date('2012-08-01');
    const TimesDom = renderer.create(<Times start={start} end={end} />)
        .toJSON();
    expect(TimesDom)
        .toMatchSnapshot();
});
