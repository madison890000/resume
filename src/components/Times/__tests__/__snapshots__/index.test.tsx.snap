// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Times 1`] = `
<div
  className="times"
>
  <span>
    Jan, 2012
  </span>
  <span>
    -
  </span>
  <span>
    Aug, 2012
  </span>
</div>
`;
